PYME/LMVis/layers/octree.py

```diff
@@ -68,106 +69,77 @@ def update_from_datasource(self, ds):
+            # Plot leaf nodes
+            nodes = nodes[(np.sum(nodes['children'],axis=1) == 0)&(nodes['depth'] > 0)]
             box_sizes = np.vstack(ds.box_size(nodes['depth'])).T
             
             alpha = nodes['nPoints']*((2.0**nodes['depth']))**3
 
-
-        # First we need the vertices of the cube. We find them from the center c provided and the box size (lx, ly, lz)
-        # provided by the octree:
-
         if len(nodes) > 0:
             c = nodes['centre']  # center
-            v0 = c + box_sizes * -1 / 2            # v0 = c - lx/2 - ly/2 - lz/2
-            v1 = c + box_sizes * [-1, -1, 1] / 2   # v1 = c - lx/2 - ly/2 + lz/2
-            v2 = c + box_sizes * [-1, 1, -1] / 2   # v2 = c - lx/2 + ly/2 - lz/2
-            v3 = c + box_sizes * [-1, 1, 1] / 2    # v3 = c - lx/2 + ly/2 + lz/2
-            v4 = c + box_sizes * [1, -1, -1] / 2   # v4 = c + lx/2 - ly/2 - lz/2
-            v5 = c + box_sizes * [1, -1, 1] / 2    # v5 = c + lx/2 - ly/2 + lz/2
-            v6 = c + box_sizes * [1, 1, -1] / 2    # v6 = c + lx/2 + ly/2 - lz/2
-            v7 = c + box_sizes / 2                 # v7 = c + lx/2 + ly/2 + lz/2
-
-            #
-            #     z
-            #     ^
-            #     |
-            #    v1 ----------v3
-            #    /|           /|
-            #   / |          / |
-            #  v5----------v7  |
-            #  |  |    c    |  |
-            #  | v0---------|-v2
-            #  | /          | /
-            #  v4-----------v6---> y
-            #  /
-            # x
-            #
-            # Now note that the counterclockwise triangles (when viewed straight-on) formed along the faces of the cube are:
-            #
-            # v0 v2 v6
-            # v0 v4 v5
-            # v1 v3 v2
-            # v2 v0 v1
-            # v3 v1 v5
-            # v3 v7 v6
-            # v4 v6 v7
-            # v5 v1 v0
-            # v5 v7 v3
-            # v6 v2 v3
-            # v6 v4 v0
-            # v7 v5 v4
-
-
-            # Concatenate vertices, interleave, restore to 3x(3N) points (3xN triangles),
-            # and assign the points to x, y, z vectors
-            triangle_v0 = np.vstack((v0, v0, v1, v2, v3, v3, v4, v5, v5, v6, v6, v7))
-            triangle_v1 = np.vstack((v2, v4, v3, v0, v1, v7, v6, v1, v7, v2, v4, v5))
-            triangle_v2 = np.vstack((v6, v5, v2, v1, v5, v6, v7, v0, v3, v3, v0, v4))
-
-            x, y, z = np.hstack((triangle_v0, triangle_v1, triangle_v2)).reshape(-1, 3).T
-
-            # Now we create the normal as the cross product (triangle_v2 - triangle_v1) x (triangle_v0 - triangle_v1)
-            triangle_normals = np.cross((triangle_v2 - triangle_v1), (triangle_v0 - triangle_v1))
-            # We copy the normals 3 times per triangle to get 3x(3N) normals to match the vertices shape
-            xn, yn, zn = np.repeat(triangle_normals.T, 3, axis=1)
+            shifts = (box_sizes[:,None]*OCT_SHIFT[None,:])*0.5
```

Comment:
Why the condensing - I don't think this aids clarity ...

Reply:
I thought I was clarifying too much with all of the comments I made. I figured this section of code didn't need to explained in such detail and fewer lines of code would give a better visual impression of the weight of its importance.

Reply:
in general more comments is always better ... just because we don't usually have enough doesn't make it a great idea to change something to visually de-emphasise it.